pgjdbc/src/test/java/org/postgresql/test/jdbc42/SetObject310Test.java

```diff
@@ -199,7 +214,7 @@ public void testSetLocalDateTime() throws SQLException {
       ZoneId zone = ZoneId.of(zoneId);
       for (String date : datesToTest) {
         LocalDateTime localDateTime = LocalDateTime.parse(date);
-        String expected = localDateTime.atZone(zone)
```

Comment:
IMHO it is correct this was changed, asserting that `2000-03-26T02:00:00` is stored or retrieved as `2000-03-26T03:00:00` when the system time zone is a specific one is wrong IMHO. `2000-03-26T02:00:00` is `2000-03-26T02:00:00`. As it has no time zone the behaviour should be the same and independent of any time zone.

I think the tests should be changed though to not call `#getString` as it currently calls `PgResultSet#internalGetObject` which ends up calling `java.sql.Timestamp#toString()` which is bound to the JVM time zone and can therefore not display timestamps that don't exist in that time zone. As `LocalDateTime` has no time zone this is wrong IMHO.

Reply:
makes sense.. care to provide a PR ?

Reply:
I'm looking at this and wondering why we are using timezones at all on LocalDateTime?

Reply:
> makes sense.. care to provide a PR ?

I can have a look. I'll have to check if it breaks any tests. It may take a few days.

Reply:
Probably not worth your time until we decide how to deal with this.